tests/scene/test_sprite_2d.h

```diff
@@ -0,0 +1,260 @@
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(0, 3));
+
+		sprite_2d->set_frame(14);
+		CHECK(sprite_2d->get_frame() == 14);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(2, 3));
+	}
+
+	SUBCASE("8x4 frames") {
+		sprite_2d->set_hframes(8);
+		sprite_2d->set_vframes(4);
+
+		CHECK(sprite_2d->get_hframes() == 8);
+		CHECK(sprite_2d->get_vframes() == 4);
+
+		sprite_2d->set_frame(0);
+		CHECK(sprite_2d->get_frame() == 0);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(0, 0));
+
+		sprite_2d->set_frame(4);
+		CHECK(sprite_2d->get_frame() == 4);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(4, 0));
+
+		sprite_2d->set_frame(8);
+		CHECK(sprite_2d->get_frame() == 8);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(0, 1));
+
+		sprite_2d->set_frame(16);
+		CHECK(sprite_2d->get_frame() == 16);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(0, 2));
+
+		sprite_2d->set_frame(31);
+		CHECK(sprite_2d->get_frame() == 31);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(7, 3));
+	}
+
+	SUBCASE("100x100 frames") {
+		sprite_2d->set_hframes(100);
+		sprite_2d->set_vframes(100);
+
+		CHECK(sprite_2d->get_hframes() == 100);
+		CHECK(sprite_2d->get_vframes() == 100);
+
+		sprite_2d->set_frame(0);
+		CHECK(sprite_2d->get_frame() == 0);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(0, 0));
+
+		sprite_2d->set_frame(60);
+		CHECK(sprite_2d->get_frame() == 60);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(60, 0));
+
+		sprite_2d->set_frame(120);
+		CHECK(sprite_2d->get_frame() == 120);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(20, 1));
+
+		sprite_2d->set_frame(240);
+		CHECK(sprite_2d->get_frame() == 240);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(40, 2));
+
+		sprite_2d->set_frame(360);
+		CHECK(sprite_2d->get_frame() == 360);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(60, 3));
+
+		sprite_2d->set_frame(2048);
+		CHECK(sprite_2d->get_frame() == 2048);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(48, 20));
+
+		sprite_2d->set_frame(8192);
+		CHECK(sprite_2d->get_frame() == 8192);
+		CHECK_EQ(sprite_2d->get_frame_coords(), Vector2i(92, 81));
+	}
+
+	memdelete(sprite_2d);
+}
+
+TEST_CASE("[SceneTree][Sprite2D] Flip testing") {
+	Sprite2D *sprite_2d = memnew(Sprite2D);
+
+	SUBCASE("Both False") {
+		CHECK_FALSE(sprite_2d->is_flipped_h());
+		CHECK_FALSE(sprite_2d->is_flipped_v());
+	}
+
+	SUBCASE("Both True") {
+		sprite_2d->set_flip_h(true);
+		sprite_2d->set_flip_v(true);
+
+		CHECK(sprite_2d->is_flipped_h());
+		CHECK(sprite_2d->is_flipped_v());
+	}
+
+	SUBCASE("True False") {
+		sprite_2d->set_flip_h(true);
+		sprite_2d->set_flip_v(false);
+
+		CHECK(sprite_2d->is_flipped_h());
+		CHECK_FALSE(sprite_2d->is_flipped_v());
+	}
+
+	SUBCASE("False True") {
+		sprite_2d->set_flip_h(false);
+		sprite_2d->set_flip_v(true);
+
+		CHECK_FALSE(sprite_2d->is_flipped_h());
+		CHECK(sprite_2d->is_flipped_v());
+	}
+
+	memdelete(sprite_2d);
+}
+TEST_CASE("[SceneTree][Sprite2D] Offset") {
```

Comment:
Add a blank line between each test case:

```suggestion
}

TEST_CASE("[SceneTree][Sprite2D] Offset") {
```